### PR TITLE
Schedule SPDX license list update

### PR DESCRIPTION
**Pull request by nobu**
## What was the end-user or developer problem that led to this PR?

"lib/rubygems/util/licenses.rb" file is often out of date.

## What is your fix for the problem, implemented in this PR?

To schedule SPDX license list update by GitHub Actions.
This weekly action will check the lists and create a PR updating the file.
Also you can dispatch the workflow manually.

## Make sure the following tasks are checked

- [x] Describe the problem / feature
- [ ] Write [tests](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#tests) for features and bug fixes
- [x] Write code to solve the problem
- [x] Make sure you follow the [current code style](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#code-formatting) and [write meaningful commit messages without tags](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#commit-messages)

### CODE REVIEW

File: .github/workflows/weekly-update.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+jobs:
+  weekly_update:
+    name: Rubygems weekly update
+    runs-on: ${{ matrix.os }}
+    if: github.repository == 'rubygems/rubygems'
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-22.04 ]
+        ruby: [ ruby-head ]
```
[the remark this answers is not in the document]
Comment (nobu) [reply .github/workflows/weekly-update.yml]: I just copied this file from another file (I don't remember well).
But noticed that the Ubuntu runner already has ruby.

**Comment by nobu** [comment]
With the system ruby: https://github.com/nobu/rubygems/actions/runs/5886920530/job/15965545155
And the result PR: https://github.com/nobu/rubygems/pull/70

**Review by hsbt — APPROVED** [review]

**Comment by hsbt** [comment]
Thank you @nobu .

**Comment by deivid-rodriguez** [comment]
Thanks @nobu!

One little tweak that just came to my mind. These PRs will always be released as "Enhancements", so we can label the generated PR with the proper label directly. Will open a follow up!

**Pull request merged**